dev-preview/README.md

```diff
@@ -199,3 +200,172 @@ The OpenShift Cluster API Operator deploys the main components (CAPI & CAPA depl
+
+curl --key /tmp/client.key --cert /tmp/client.crt  -H "Content-Type: application/json" --data "@data/k8s-cluster.json" --cacert /tmp/ca.crt https://$inventory_api_route:443/api/inventory/v1beta1/resources/k8s-clusters
+
+cat data/k8s-cluster.json
+
+{
+ "k8s_cluster": {
+   "metadata": {
+     "resource_type": "k8s-cluster",
+     "workspace": ""
+   },
+   "reporter_data": {
+     "reporter_type": "ACM",
+     "reporter_instance_id": "guest",
+     "reporter_version": "0.1",
+     "local_resource_id": "1",
+     "api_href": "www.example.com",
+     "console_href": "www.example.com"
+   },
+   "resource_data": {
+     "external_cluster_id": "1234",
+     "cluster_status": "READY",
+     "kube_version": "1.31",
+     "kube_vendor": "OPENSHIFT",
+     "vendor_version": "4.16",
+     "cloud_platform": "AWS_UPI",
+     "nodes": [
+       {
+         "name": "www.example.com",
+         "cpu": "7500m",
+         "memory": "30973224Ki",
+         "labels": [
+           {
+             "key": "has_monster_gpu",
+             "value": "yes"
+           }
+         ]
+       }
+     ]
+   }
+ }
+}
+```
+The cluster information should be sent to Kafka topic. We can use `kafka-console-consumer.sh` to check the results.
+```
+caPassword=`kubectl get secret kafka-cluster-ca-cert -o jsonpath='{.data.ca\.password}' | base64 -d`
+kubectl get secret kafka-cluster-ca-cert -o jsonpath='{.data.ca\.p12}' | base64 -d > /tmp/ca.p12
+kubectl get secret global-hub-kafka-user -o jsonpath='{.data.user\.p12}' | base64 -d > /tmp/user.p12
+userPassword=`kubectl get secret  global-hub-kafka-user -o jsonpath='{.data.user\.password}' | base64 -d`
+kubectl cp /tmp/user.p12 kafka-kafka-0:/tmp
+kubectl cp /tmp/ca.p12 kafka-kafka-0:/tmp
+
+cat << EOF > /tmp/client.properties
+security.protocol=SSL
+ssl.truststore.location=/tmp/ca.p12
+ssl.truststore.password=${caPassword}
+ssl.keystore.location=/tmp/user.p12
+ssl.keystore.password=${userPassword}
+EOF
+kubectl cp /tmp/client.properties kafka-kafka-0:/tmp
+
+# access pod to check topics
+oc exec -it kafka-kafka-0 sh
+bin/kafka-console-consumer.sh --bootstrap-server $kafka_bootstrap_server:443 --consumer.config=/tmp/client.properties --topic kessel-inventory --from-beginning
+```
+The result looks like this:
+```
+{"metadata":{"id":0,"last_reported":"2024-09-11T15:45:08.941075631Z","resource_type":"k8s-cluster","workspace":"","labels":null},"reporter_data":{"reporter_instance_id":"guest","reporter_type":"ACM","last_reported":"2024-09-11T15:45:08.941075631Z","local_resource_id":"1","reporter_version":"0.1","console_href":"www.example.com","api_href":"www.example.com"},"resource_data":{"external_cluster_id":"1234","cluster_status":"READY","kube_version":"1.31","kube_vendor":"OPENSHIFT","vendor_version":"4.16","cloud_platform":"AWS_UPI","nodes":[{"name":"www.example.com","cpu":"7500m","memory":"30973224Ki","labels":[{"key":"has_monster_gpu","value":"yes"}]}]}}
+```
+Congratulations! You have set up the environment correctly.
+
```

Comment:
A Red Hat management tool like ACM would have automatically registered its cluster and policies etc using this API. In that case, you would directly see the events without having to make the API POST call.